code/game/objects/effects/spawners/abnormality_room.dm

```diff
@@ -25,11 +25,13 @@
 	for(var/obj/machinery/computer/abnormality/AC in allObjects)
 		AC.datum_reference = abno_datum
 		break
+	for(var/area/containment_zone/ACR)
+		ACR.name = "CZ: [abno_datum.name] ([THREAT_TO_NAME[abno_datum.threat_level]])"
```

Comment:
I assume you didn't test this much, huh? You are iterating through nothing.
You should simply use get_area() proc to get the area in place and then work with it.

Reply:
Oups yeah I see it now it stay on the first abno that spawn and keep the name aaaa